src/main/java/org/pawaneuler/Generator/TransactionTrieGenerator/TransactionTrieCreator.java

```diff
@@ -27,81 +27,64 @@ public Trie createTranssactionTrie() throws IOException {
         String[] products = tcsvData.getProducts();
 
         while (products != null) {
-            // traverse the trie from the root node
-            int trav = 0;
-            int productsLength = products.length;
+            this.insertProduct(trie, products);
 
-            for (int i = 0; i < productsLength; i++) {
-                Node currentNode = trie.getNodeAt(trav);
+            tcsvData = source.readLine();
+            products = tcsvData.getProducts();
+        }
+
+        return trie;
+    }
+
+    private void insertProduct(Trie trie, String[] products) {
+        // traverse the trie from the root node
+        int trav = 0;
+        int productsLength = products.length;
+
+        for (int i = 0; i < productsLength; i++) {
```

Comment:
Non descriptive name of `i`
```suggestion
        for (int currentProductIndex = 0; i < productsLength; currentProductIndex++) {
```

Reply:
I think i is okay

Reply:
A bit too vague, but heh, I'll pass it lol